VideoLocker/src/main/java/org/edx/mobile/util/FileUtil.java

```diff
@@ -38,4 +41,25 @@
             inputStream.close();
         }
     }
+
+    /**
+     * Deletes a single file and/or recursively deletes files/folders within a folder before
+     * deleting it.
+     *
+     * @param fileOrDirectory The file/folder that needs to be deleted.
+     * @param exceptions      Name of the file/folder that needs to be skipped while deletion.
+     */
+    public static void deleteRecursive(File fileOrDirectory, List<String> exceptions) {
+        if (exceptions.contains(fileOrDirectory.getName())) return;
+
+        if (fileOrDirectory.isDirectory()) {
+            File[] filesList = fileOrDirectory.listFiles();
+            if (filesList != null) {
+                for (File child : filesList) {
+                    deleteRecursive(child, exceptions);
+                }
+            }
+        }
+        fileOrDirectory.delete();
```

Comment:
There is a warning here for ignoring the result of this method. I suppose that behaviour is fine in this case, since we don't want to short-circuit the whole recursive operations on encountering one error. You can suppress it by adding `//noinspection ResultOfMethodCallIgnored` before this line.